### DISCUSSION TITLE
Change in Endpoint class not get in client side : serverpod

### DISCUSSION THREAD

**Discussion by itsdani121** [serverpod/serverpod#661] [Q&A] [answered]
In ExampleEndpoint class i has created a demoTesting function which then working fine but after change the name and add some other function. when i command "serverpod generate --watch" it works fine but on client side it did not show any changes. At that time server is also running "dart bin/main.dart" with this command the previous data came from server to client side perfectly but when i changes it don't appear in client side.

            ` 
             //Server side class

           class ExampleEndpoint extends Endpoint {
  
       Future<List<Testing>> getTesting(Session session,
       {String? keyword, int? id}) async {
            return Testing.find(
       session,
       where: (p0) =>
           keyword != null ? p0.title.like("%$keyword%") : Constant(true),
          );
          }

            Future<bool> addTesting(Session session, Testing testing) async {
           await Testing.insert(session, testing);
           return true;
            }

              Future<bool> updateTesting(Session session, Testing testing) async {
                await Testing.update(session, testing);
               return true;
                 }

               Future<bool> deleteTesting(Session session, int id) async {
              await Testing.delete(
                                   session,
                                   where: (p0) => p0.id.equals(id),
                                     );
                             return true;
                              }
                              }
                  

                          //client side response 
                         
                        //Warning: The method 'getTesting' isn't defined for the type '_EndpointExample'. (Documentation)  Try correcting the   
             name to the name of an existing method, or defining a method named 'getTesting'.
                           
                         _testing = await client.example.getTesting(); 
                                  print('test result is $_testing');
 

            `

   **Serverpod versions
               -  Serverpod version: 1.0.0
            _Library version:dependencies:
                       serverpod: ^1.0.0.

**Platform information**
Android 10 Huawai honor 8x

**Additional context**
Also how i get data in client class using fromjson method because it want "serializationManager" what is this. and how to use it.
actually i am new in serverpod backend not know about these things even documentation not explain clearly about it.

**Comment by jijkbird** [comment]
I also have this problem

**Comment by vlidholt** [comment]
It is possible that you need to restart the Dart analyzer to see the changes on the client side. If you are using VS Code you can do this by `Cmd-Shift-P` then `Dart: Restart analysis server`.

**Reply by itsdani121** [reply]
actually i am using vs code for server side and android studio for client side. so what happens when i do this. also when i change in my Endpoint class it need to be restart?

**Comment by Saiprasadlokhande** [comment, marked as answer]
hi @itsdani121 
For this scenario, just try the `flutter pub get` command in the flutter app once you did changes on the server side.
this will save you time by not restarting the studio again and again.

I guess the changes are reflected in client package, but flutter_app needs the above command to fetch those changes. 
Give it a try.

**Reply by itsdani121** [reply]
yes , it works now but why it need to be do that, can you tell me about  last part of my question?which is additional context part
@Saiprasadlokhande

**Reply by Saiprasadlokhande** [reply]
@itsdani121 
We are able to use client code because, as you will see in your flutter_app --> pubspec.yaml, you have added your client code as dependency, and to get current_updates of that dependency we call `flutter pub get`.

I am not 100% sure, this is my understanding. 
But I hope you will get idea about how this works.

**Reply by itsdani121** [reply]
@Saiprasadlokhande what about additional context ?

**Reply by itsdani121** [reply]
i also got error when calling update method here is ServerpodClientException: Internal server error. Call log id: 0, statusCode = 500.
i dont know why it give me error even all other method working fine
